src/mlpack/tests/lrsdp_test.cpp

```diff
@@ -118,6 +118,63 @@ BOOST_AUTO_TEST_CASE(Johnson844LovaszThetaSDP)
   }
 }
 
+BOOST_AUTO_TEST_CASE(GaussianMatrixSensingSDP)
+{
+  arma::mat Xorig, A, bmat;
+
+  data::Load("sensing_X.csv", Xorig, true, false);
+  data::Load("sensing_A.csv", A, true, false);
+  data::Load("sensing_b.csv", bmat, true, false);
+
+  arma::vec b = bmat;
+
+  const size_t m = Xorig.n_rows;
+  const size_t n = Xorig.n_cols;
+  const size_t p = b.n_elem;
+
+  assert(A.n_rows == p);
+  assert(A.n_cols == m * m);
+
+  float r = 0.5 + sqrt(0.25 + 2 * p);
+  if (ceil(r) > m + n)
+    r = m + n;
+
+  std::srand(0x0FAE8543);
```

Comment:
I've done the same sin here as in #372. Let me update this with a deterministic starting point.